src/mongocxx/database.cpp

```diff
@@ -157,5 +164,12 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const std::unique_ptr<database::impl>& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
```

Comment:
Do we want to use operation_exception? Also, will it work to use the same type of exception in all cases?